src/docfx/lib/HrefUtility.cs

```diff
@@ -98,11 +100,42 @@ public static DependencyType FragmentToDependencyType(string fragment)
             return fragment != null && fragment.Length > 1 ? DependencyType.Bookmark : DependencyType.Link;
         }
 
-        public static bool IsAbsoluteHref(string str)
+        public static HrefType GetHrefType(string href)
         {
-            return str.StartsWith('/')
-                || str.StartsWith('\\')
-                || Uri.TryCreate(str, UriKind.Absolute, out _);
+            if (string.IsNullOrEmpty(href))
+            {
+                return HrefType.RelativePath;
+            }
+
+            var ch = href[0];
+
+            if (ch == '/' || ch == '\\')
+            {
+                return HrefType.AbsolutePath;
+            }
+
+            // If it is a windows rooted path like C:
+            if (href.Length > 2 && href[1] == ':')
+            {
+                return HrefType.AbsolutePath;
+            }
+
+            if (Uri.TryCreate(href, UriKind.Absolute, out _))
+            {
+                return HrefType.External;
+            }
+
+            if (char.IsLetter(ch) && href.Contains(':'))
+            {
+                return HrefType.External;
+            }
```

Comment:
what's is this for if we already have Uri.TryCreate above?

Reply:
To cover the test case
```csharp
[InlineData("feedback-url:?query=a", HrefType.External)]
```

Reply:
is this a url with port?

Reply:
this will launch a windows application